### PR TITLE
feat: Deprecate from chectl tslint and add eslint with check license headers

### PR DESCRIPTION
**Pull request by flacatus**
Signed-off-by: Flavius Lacatusu <flacatus@redhat.com>

### What does this PR do?
This PR deprecate tslint and add support for eslint. Also this PR add eslint license header checks.

Refference issue: eclipse/che#19647

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

N/A

### How to test this PR?

N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by benoitf — APPROVED** [review]
LGTM,

just finding the `if {} if {} if {}` harder than the `if else if`

File: package.json [hunk 1]
```diff
@@ -146,26 +151,26 @@
   },
   "repository": "che-incubator/chectl",
   "scripts": {
-    "postinstall": "npm run -s postinstall-repositories && npm run -s postinstall-helm && npm run -s postinstall-cert-manager && npm run -s postinstall-operator && npm run -s postinstall-dev-workspace && npm run -s postinstall-cleanup",
+    "postinstall": "yarn lint && npm run -s postinstall-repositories && npm run -s postinstall-helm && npm run -s postinstall-cert-manager && npm run -s postinstall-operator && npm run -s postinstall-dev-workspace && npm run -s postinstall-cleanup",
```
Comment (mmorhun) [inline package.json:154]: Do we really need to run `lint` on each `install` ?

File: src/api/che-login-manager.ts [hunk 2]
```diff
@@ -1,12 +1,14 @@
-/*********************************************************************
- * Copyright (c) 2020 Red Hat, Inc.
- *
+/**
+ * Copyright (c) 2019-2021 Red Hat, Inc.
```
Comment (mmorhun) [inline src/api/che-login-manager.ts:2]: Shouldn't it be `2020-2021`?

File: src/api/che-login-manager.ts [hunk 3]
```diff
@@ -185,7 +190,7 @@ export class CheServerLoginManager {
 
     const allLogins = new Map<string, string[]>()
     for (const [apiUrl, serverLogins] of Object.entries(this.loginData.logins!)) {
-      allLogins.set(apiUrl, Array.from(Object.keys(serverLogins)))
+      allLogins.set(apiUrl, [...Object.keys(serverLogins)])
```
Comment (mmorhun) [inline src/api/che-login-manager.ts:193]: What is the purpose of this change?

Comment (flacatus) [reply src/api/che-login-manager.ts:193]: @mmorhun This change was forced by https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v6.0.1/docs/rules/prefer-spread.md... we can add a eslint comment if you wish

File: src/api/che-login-manager.ts (hunk 2 above)
Comment (flacatus) [reply src/api/che-login-manager.ts:2]: Old files should be 2019-2021
New files 2021-2021?

Comment (flacatus) [reply src/api/che-login-manager.ts:2]: This file was changed automatically by eslint

File: .eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+ /**
+ *
```
Comment (tolusha) [inline .eslintrc.js:2]: Extra line

**Review by tolusha — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (flacatus) [reply package.json]: My bad. I added to every chectl binaries pack.
